PWGDQ/Tasks/tableReader.cxx

```diff
@@ -351,6 +363,196 @@ struct AnalysisMuonSelection {
   PROCESS_SWITCH(AnalysisMuonSelection, processDummy, "Dummy function", false);
 };
 
+struct AnalysisPrefilterSelection {
+  Produces<aod::Prefilter> prefilter;
+  Preslice<MyBarrelTracks> perCollision = aod::reducedtrack::reducedeventId;
+
+  // Configurables
+  Configurable<std::string> fConfigDalitzTrackCutsId{"cfgDalitzTrackCutsId", "", "Id of Dalitz track selection cuts (from cfgTrackCuts in AnalysisTrackSelection, starting at 0 = first dalitz cut)"};
+  Configurable<std::string> fConfigDalitzPairCuts{"cfgDalitzPairCuts", "", "Dalitz pair selection cuts"};
+  Configurable<std::string> fConfigAddTrackPairHistogram{"cfgAddTrackPairHistogram", "", "Comma separated list of histograms"};
+  Configurable<bool> fQA{"cfgQA", false, "QA histograms"};
+
+  Filter filterBarrelTrackSelected = aod::dqanalysisflags::isBarrelSelectedDalitz > 0; // Dalitz candidates
+
+  OutputObj<THashList> fOutputList{"output"}; //! the histogram manager output list
+  OutputObj<TList> fStatsList{"Statistics"};  //! skimming statistics
+
+  std::map<int, uint8_t> dalitzmap;
+
+  std::vector<int> fTrackCutsId;
+  std::vector<AnalysisCompositeCut> fPairCuts;
+  int nCuts = 0;
+
+  HistogramManager* fHistMan;
+
+  void init(o2::framework::InitContext& context)
+  {
+    // Barrel track cuts
+    TString cutNamesStr = fConfigDalitzTrackCutsId.value;
+    if (!cutNamesStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(cutNamesStr.Tokenize(","));
+      for (int icut = 0; icut < objArray->GetEntries(); ++icut) {
+        fTrackCutsId.push_back(std::stoi(objArray->At(icut)->GetName()));
+      }
+    }
+
+    // Pair cuts
+    TString cutNamesPairStr = fConfigDalitzPairCuts.value;
+    if (!cutNamesPairStr.IsNull()) {
+      std::unique_ptr<TObjArray> objArray(cutNamesPairStr.Tokenize(","));
+      for (int icut = 0; icut < objArray->GetEntries(); ++icut) {
+        fPairCuts.push_back(*dqcuts::GetCompositeCut(objArray->At(icut)->GetName()));
+      }
+    }
+
+    if (fTrackCutsId.size() != fPairCuts.size()) {
+      std::cout << "WARNING: YOU SHOULD PROVIDE THE SAME NUMBER OF TRACK AND PAIR CUTS" << std::endl;
+    }
+    nCuts = std::min(fTrackCutsId.size(), fPairCuts.size());
+
+    VarManager::SetUseVars(AnalysisCut::fgUsedVars); // provide the list of required variables so that VarManager knows what to fill
+    VarManager::SetDefaultVarNames();
+    fHistMan = new HistogramManager("analysisHistos", "aa", VarManager::kNVars);
+    fHistMan->SetUseDefaultVariableNames(kTRUE);
+    fHistMan->SetDefaultVarNames(VarManager::fgVariableNames, VarManager::fgVariableUnits);
```

Comment:
if the fQA configurable is false, you don't need to initialize the var manager anymore. The same with the other blocks below. If no QA is needed, we should not create histograms or output directories.